src/jobs/run.yml

```diff
@@ -40,6 +40,10 @@ parameters:
     type: boolean
     default: false
 
+  pipeline_id:
+    description: The CircleCI Pipeline ID ( << pipeline.id >> ), used to rerun only failed tests when a CircleCI workflow is rerun
+    type: string
+
```

Comment:
Could this actually just be a boolean? The "<< pipeline.id >>" value is the only valid option (I believe?), so we may instead ask the user if the want to re-run only failed tests.

Reply:
Alternatively, this could be an enum value with distinct possible values.
https://circleci.slack.com/archives/C48NG2XFW/p1594214447393500

Something like "all, only failed"

Reply:
@KyleTryon I'm guessing that's not the right link? We need the user to explicitly pass their `<< pipeline.id >>` to the orb in the config file though, right? Since we don't have access to it within the orb otherwise.

Reply:
Oops sorry for missing this and correct that was not the right link haha. This may have to be tested as I am not sure the specific order but the orb is simply config, therefore writing it within your orb, is no different than the user writing it themselves. At the time the pipeline is executed the `<< pipeline.id >>` within the config should be translated. Hope this makes sense.

@magni-

Reply:
Hmm, referencing `<< pipeline.id >>` directly within the command steps leads to validation errors:
```sh
$ circleci config pack src/ > orb.yml && circleci orb validate orb.yml
Error: Error calling command: 'run_qa'
Unknown variable(s): id
Error calling job: 'run'
Error calling command: 'run_qa'
Unknown variable(s): id
```

When I reference it in the command parameters (as a default value), `orb.yml` is considered valid, but the parameter doesn't end up getting set:
```yml
# commands/run_qa.yml source
parameters:
  # ...
  pipeline_id:
    type: string
    default: << pipeline.id >>

steps:
  # ...
  run:
    command: |
      # ...
      # Check for rerun
        if [ -n "<< parameters.pipeline_id >>" ] && [ -s ~/pipeline/<< parameters.pipeline_id >> ] ; then
```
```sh
# Circle build output:
# Check for rerun
if [ -n "" ] && [ -s ~/pipeline/ ] ; then
```

Reply:
hmmm interesting. Thank you for testing. I suppose the pipeline injection must happen first, which may actually make sense, as orbs are translated at execution time and the pipeline starts prior to that. Looks like you have the best option in place 👍